analyzer/block/block.go

```diff
@@ -39,7 +39,9 @@ const (
 // block based analyzer.
 type BlockProcessor interface {
 	// PreWork performs tasks that need to be done before the main processing loop starts.
+	// When running in parallel mode, this method will only be called once.
```

Comment:
more info: when running in parallel mode with multiple analyzer instances, only one of those instances will have PreWork called. so don't use it to initialize any fields of the processor. only use it for shared state, such as fixing up the database